src/apps/statements/service/storeStatements/updateFullActivities.ts

```diff
@@ -43,79 +92,83 @@ export interface Opts {
-    }).filter((value) => {
-      return value !== undefined;
-    });
-
-    return {
-      activityId,
-      name: Object.assign({}, ...names),
-      description: Object.assign({}, ...descriptions),
-      extensions: Object.assign({}, ...extensions),
-      ...(types.length > 0 ? { type: types[types.length - 1] } : {}),
-      ...(moreInfos.length > 0 ? { moreInfo: moreInfos[moreInfos.length - 1] } : {}),
-    };
-  });
-
-  await config.repo.updateFullActivities({ updates, client });
-  // tslint:disable-next-line:max-file-line-count
+  const groupedActivities = groupBy(
+    definedActivities,
+    (activity) => activity.activityId,
+  );
+
+  const fullActivities = map(
+    groupedActivities,
+    (matchingActivities, activityId): FullActivityDatabase => {
+      const names = matchingActivities.map(
+        (matchingActivity) => matchingActivity.name,
+      );
+
+      const descriptions = matchingActivities.map(
+        (matchingActivity) => matchingActivity.description,
+      );
+
+      const extensions = matchingActivities.map(
+        (matchingActivity) => matchingActivity.extensions,
+      );
+
+      const contextActivitiesForFullActivity: FullActivityContextActivities[] =
+        matchingActivities.map((matchingActivity) =>
+          mapValues(matchingActivity.context?.contextActivities, (matchingContextActivities) =>
+            matchingContextActivities !== undefined
+              // tslint:disable-next-line:max-line-length
+              ? matchingContextActivities.map((matchingContextActivity) => matchingContextActivity.id)
+              : matchingContextActivities,
+          ));
+
+      const mergedContextActivities = contextActivitiesForFullActivity.reduce(
+        (acc, contextActivity) => {
+
+          return mergeWith(acc, contextActivity, (objValue, srcValue) =>
+            isArray(objValue) ? uniq(objValue.concat(srcValue)) : objValue);
+        },
+      );
+
+      const type = last(
+        matchingActivities
+        .map((matchingActivity) => matchingActivity.type)
+        .filter((value) => value !== undefined),
+      );
+
+      const moreInfo = last(
+        matchingActivities
+        .map((matchingActivity) => matchingActivity.moreInfo)
+        .filter((value) => value !== undefined),
+      );
+
+      return {
+        activityId,
+        lrsId: client.lrs_id,
+        organisationId: client.organisation,
+        name: Object.assign({}, ...names),
+        description: Object.assign({}, ...descriptions),
+        extensions: Object.assign({}, ...extensions),
+        ...(isEmpty(mergedContextActivities)
+          ? {}
+          : { context: { contextActivities: mergedContextActivities } }),
+        ...(type !== undefined ? { type } : {}),
+        ...(moreInfo !== undefined ? { moreInfo } : {}),
+      };
+    },
+  );
+
+  await config.repo.updateFullActivities({ fullActivities });
+// tslint:disable-next-line:max-file-line-count
```

Comment:
@PrinceWaune we don't use such ignore lines in other modules within the xapi-service - is this required?